### DISCUSSION TITLE
Difference from goutubedl

### DISCUSSION THREAD

**Discussion by QuartzWarrior** [lrstanley/go-ytdlp#14] [Q&A] [answered]
What is the difference between this repo and [goutubedl](https://github.com/wader/goutubedl)? As far as I can tell they both aim to do the same thing, one just has more stars and users than the other. 

( For more background, I searched "ytdlp golang" and both of these popped up, looking for nice speed really, although yt-dlp uses python )

**Comment by lrstanley** [comment, marked as answer]
Will say both wrap yt-dlp, so you can't get away from python completely (though see pt `3`). However, I will say:
1. goutubedl looks to be for relatively simple usecases.
2. go-ytdlp is generated based off yt-dlp directly, and as such includes support for all of the flags/parameters that yt-dlp provides. The downside with this is it may not be as polished around the edges.
3. go-ytdlp handles installing and caching the appropriate version of yt-dlp behind the scenes (and securely). It uses the binary variant of yt-dlp, so you only need to pull in the library, and it handles everything else for you.

Other than the above, I don't have experience with using that library directly, so can't say what else may be different.

**Reply by QuartzWarrior** [reply]
Yeah I was noticing how goutubedl has a lot fewer options compared to this lib. Thanks
